### PR TITLE
Make the `Preview` parse the `previewTree` state

### LINKED ISSUE #39: Make the `Preview` view display a parsed  version of the `previewTree`

### PR DESCRIPTION
**Pull request by YacineSteeve**
* Closes #39
<hr>

Here is the rendering based on the update `previewTree` from the `arborescence.ts` file:<br>

It uses 5 test cases based on components with different children number (0 or many)

⚠️  The great number of changed files in mainly due to `Prettier` formatting (I increase the line line fro `80` to `100`).
The files of interest, needing a review, are only 9:
- `apps/frontend/src/views/Preview/Preview.tsx`
- `apps/frontend/src/views/Preview/arboresence.ts`
- `apps/frontend/src/libs/types/tree.type.ts`
- `apps/frontend/src/libs/types/global.type.ts`
- `apps/frontend/src/components/SideBarRight/SideBarRight.tsx`
- `packages/types/index.ts`
- `packages/ui/src/components/Button/Button.tsx`
- `packages/ui/src/components/Button2/Button2.tsx`
- `packages/ui/src/main.tsx`

### CODE REVIEW

**Comment by YacineSteeve** [comment]
Current dependencies on/for this PR:
* main
  * **PR #48** <a href="https://app.graphite.dev/github/pr/Frelya/react-site-editor/48" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/Frelya/react-site-editor/48?utm_source=stack-comment).

File: apps/frontend/src/libs/types/property.type.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import type { PropType } from '@react-site-editor/types';
+
+export interface PropertyBaseInterface {
```
Comment (YacineSteeve) [inline apps/frontend/src/libs/types/property.type.ts:3]: Don't pay attention to this file ^^ '

File: packages/types/index.ts [hunk 2]
```diff
@@ -1,11 +1,20 @@
-export interface Component<T> {
+export interface PredefinedComponent {
```
Comment (epoundor) [inline packages/types/index.ts:1]: PredefinedComponent must be a generic.
And please stop put any in your code

File: packages/types/index.ts [hunk 3]
```diff
@@ -1,11 +1,20 @@
...
     caller: React.FunctionComponent;
-    defaultProps: T;
+    defaultProps: PredefinedComponentProps;
+}
+
+export interface PredefinedComponentProps {
```
Comment (epoundor) [inline packages/types/index.ts:6]: Must be PredefinedComponentProps<T> and defaultProps must inherited T

File: apps/frontend/src/libs/types/global.type.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export type SideBarScale = '1' | '2';
```
Comment (epoundor) [inline apps/frontend/src/libs/types/global.type.ts:1]: I don't know what your mean.
Must be an Enum in the better case

File: apps/frontend/src/components/SideBarRight/SideBarRight.tsx [hunk 5]
```diff
@@ -22,6 +22,10 @@ const PROPERTY_COMPONENTS_MAP: Record<string, React.FunctionComponent<any>> = {
     SIZE: SizeProperty
 };
 
+function isComponentProp(prop: object): prop is ComponentProp {
```
Comment (epoundor) [inline apps/frontend/src/components/SideBarRight/SideBarRight.tsx:25]: I don't understand what you mean

**Review by epoundor — CHANGES_REQUESTED** [review]

File: apps/frontend/src/components/SideBarRight/SideBarRight.tsx (hunk 5 above)
Comment (YacineSteeve) [reply apps/frontend/src/components/SideBarRight/SideBarRight.tsx:25]: It's called a type guard.
1. the function is called with `prop` of type `object`.
2. If the condition is satisfied, it returns `true`, and change the type of `prop` to `ComponentProp`
3. Else, it only returns `false` without any change

File: apps/frontend/src/libs/types/global.type.ts (hunk 4 above)
Comment (YacineSteeve) [reply apps/frontend/src/libs/types/global.type.ts:1]: Oups, don't mind it. To be removed.

**Review by epoundor — APPROVED** [review]

**Pull request merged**